appcues/src/main/java/com/appcues/ui/primitive/HorizontalStackPrimitive.kt

```diff
@@ -49,15 +51,22 @@ internal fun HorizontalStackPrimitive.Compose(modifier: Modifier) {
 @Composable
 private fun RowScope.ItemBox(
     distribution: ComponentDistribution,
-    style: ComponentStyle,
+    primitive: ExperiencePrimitive,
     parentVerticalAlignment: Alignment.Vertical,
     content: @Composable BoxScope.() -> Unit
 ) {
     Box(
         modifier = Modifier
-            .align(style.getVerticalAlignment(parentVerticalAlignment))
-            .then(if (distribution == ComponentDistribution.EQUAL) Modifier.weight(1f) else Modifier),
-        contentAlignment = style.getBoxAlignment(),
+            .align(primitive.style.getVerticalAlignment(parentVerticalAlignment))
+            .then(
```

Comment:
this is where the magic is happening rather than in SpacerPrimitive composition.

we are checking if the ItemBox is rendering a Spacer as a Row item, if so we check if width is null, and then we apply weight (weight can only be applied to a child of a Row (in a RowScope), this is the reason this needs to be done here

Reply:
in the case of a vertical spacer, where some height is defined (if we ever allow that in the builder) - this would also work as it would just span available width (maybe full row) and then apply whatever height is defined as part of the SpacerPrimitive composition itself? Is that correct?